pkg/leader/leader.go

```diff
@@ -58,7 +58,24 @@ func run(ctx context.Context, namespace, name string, client kubernetes.Interfac
 				go cb(ctx)
 			},
 			OnStoppedLeading: func() {
-				logrus.Fatalf("leaderelection lost for %s", name)
+				select {
+				case <-ctx.Done():
```

Comment:
Would this block if the context is not done, eg., when leader election was lost for any reason other than SIGTERM?

Reply:
@moio It will not block. If the channel doesn't have something to read then that case isn't chosen. You can use select statements to look at multiple channels. You can read about it in a place more like https://hackthedeveloper.com/golang-select-multiple-channels/

Reply:
Great, I learnt something new today, thanks!